src/components/Input/index.ts

```diff
@@ -0,0 +1,45 @@
+import Block from '../../utils/Block';
+import template from './input.hbs';
+import validateValueByName from '../../utils/Validate';
+import * as styles from './input.module.scss';
+
+interface InputProps {
+  label?: string;
+  type?: string;
+  name?: string;
+  placeholder?: string;
+  events?: {
+    blur?: () => void;
+  };
+}
+
+export default class Input extends Block<InputProps> {
+  init() {
+    this.setProps({
+      events: {
+        blur: this.validateOnBlur.bind(this),
+      },
+    });
+  }
+
+  render() {
+    return this.compile(template, { ...this.props, styles });
+  }
+
+  public isValid(): boolean {
+    const inputName = this.getName();
+    return inputName !== '' ? validateValueByName(this.getValue(), inputName) : true;
+  }
+
+  public getName(): string {
+    return this.props.name!;
+  }
+
+  public getValue(): string {
+    return (this.getContent() as HTMLInputElement).value;
```

Comment:
Лучше стараться не использовать явное приведение типов там, где этого можно избежать. Отличным способ для этого являются type parameters.  Например, querySelector можно вызывать вот так
```
document.querySelecor<HTMLInputElement>(selector)
```
чтобы явно не приводить найденный результат к ```HTMLInputElement```

Тут можно сделать так же
```suggestion
    return this.getContent<HTMLInputElement>().value;
```